LogIn/LoginViewModel.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  LoginViewModel.swift
+//  LogIn
+//
+//  Created by Aleksei Permiakov on 08.12.2021.
+//
+
+import UIKit
+import Combine
+
+struct LoginViewModelInput {
+    let email: AnyPublisher<String, Never>
+    let pass: AnyPublisher<String, Never>
+    let passAgain: AnyPublisher<String, Never>
+}
+
+struct LoginViewModelOutput {
+    let emailTint: UIColor
+    let passwTint: UIColor
+    let passwAgainTint: UIColor
+    let emailFormatMessage: String
+    let isEnabled: Bool
+}
+
+protocol LoginViewModelType {
+    var outputPublisher: Published<LoginViewModelOutput>.Publisher { get }
+    func transform(input: LoginViewModelInput)
+}
+
+class LoginViewModel: LoginViewModelType {
```

Comment:
Why it is a `class` and why it is not `final`?

Reply:
made it final. Class to be able to use @Published wrapper